File: suites/05_ml_primitives/comparisons/autodiff.go
package main
import "fmt"
func autodiffTrace(steps int64) int64 {
	value, grad := int64(7), int64(1)
	for s := int64(0); s < steps; s++ { grad += (value * 3) % 17; value += grad }
	return value + grad
}
func main() { fmt.Println(autodiffTrace(5000000)) }

File: suites/05_ml_primitives/comparisons/autodiff.rs
fn autodiff_trace(steps: i64) -> i64 {
    let (mut value, mut grad) = (7i64, 1i64);
    for _ in 0..steps { grad += (value * 3) % 17; value += grad; }
    value + grad
}
fn main() { println!("{}", autodiff_trace(5000000)); }
